validator/client/validator.go

```diff
@@ -1055,15 +1061,34 @@ func (v *validator) buildProposerSettingsRequests(ctx context.Context, pubkeys [
 			})
 		}
 		if !skipAppendToFeeRecipientArray && enableValidatorRegistration {
-			registerValidatorRequests = append(registerValidatorRequests, &ethpb.ValidatorRegistrationV1{
-				FeeRecipient: feeRecipient[:],
-				GasLimit:     gasLimit,
-				Timestamp:    uint64(time.Now().UTC().Unix()),
-				Pubkey:       pubkeys[i][:],
-			})
+			signedReg, ok := v.signedValidatorRegistrations[pubkeys[i]]
+			if ok &&
+				signedReg.Message.GasLimit == gasLimit &&
+				hexutil.Encode(signedReg.Message.FeeRecipient) == hexutil.Encode(feeRecipient[:]) {
+				signedRegisterValidatorRequests = append(signedRegisterValidatorRequests, signedReg)
+			} else {
+				reg := &ethpb.ValidatorRegistrationV1{
```

Comment:
All this nested logic is complex and hard to reason about. Can we simplify conditionals with less nesting?
Also, the check for whether or not we should append to the signed requests has high cognitive complexity. It would be ideal to refactor that specific conditional into its own function that can be thoroughly unit tested for all its logic branches

Reply:
will try to rework this into something more managable